app/controllers/concerns/dateable.rb

```diff
@@ -6,8 +6,9 @@ module Dateable
   private
 
   def set_dates
-    dateRange = params[:date] ? CGI.escapeHTML(params[:date]) : "7d"
-    @start_date = case dateRange
+    @date_range = params[:date] ? CGI.escapeHTML(params[:date]) : "7d"
```

Comment:
I wanted to put this "set_date_range" into an own function above this one because it got too long, but then there were some problems with the tests... so I moved it back inside. Also, I had a shorter version for this, but it seemed to be too complex, that's why I divided it into two.

Reply:
Why do you need the `@date_range` variable here? I think it would be better to just default the `case` statement below to `6.days.ago` and add a little comment to make sure to explain that's what's happening